foubus.py

```diff
@@ -42,411 +53,552 @@
+        resp.release_conn()
+        return
 
-            resp.release_conn()
+    # New file available
+    if resp.status == 200:
+        last_modified = time.mktime(
+            time.strptime(resp.headers["Last-Modified"], "%a, %d %b %Y %H:%M:%S GMT")
+        )
 
-            os.utime(f.name, (last_modified, last_modified))
-            os.rename(f.name, os.path.basename(url))
-            logging.info("Saved to %s", os.path.basename(url))
+        with tempfile.NamedTemporaryFile(
+            dir=".", prefix=os.path.basename(url) + "-", delete=False
+        ) as tmp_file:
+            logging.info("Downloading to %s", tmp_file.name)
+            while chunk := resp.read(1024 * 1024):
+                tmp_file.write(chunk)
 
-            revalidated = datetime.datetime.now()
-        else:
-            raise ValueError(f"Unexpected status: {resp.status}")
+        resp.release_conn()
 
+        # Update mtime of new file
+        os.utime(tmp_file.name, (last_modified, last_modified))
+        os.rename(tmp_file.name, os.path.basename(url))
+        logging.info("Saved to %s", os.path.basename(url))
 
-def build_stop_timetable(date):
-    """Run at 6am"""
+        revalidated = datetime.datetime.now()
+    else:
+        resp.release_conn()
+        raise ValueError(f"Unexpected status: {resp.status}")
+
+
+def build_stop_timetable(req_date: datetime.date) -> None:
+    """
+    Builds the stop timetable for all stops listed in STOPS for the given date.
+    The result is saved to a 'stop_timetable/' directory.
+    """
     logging.info("Reading feed...")
     feed = gtfs_kit.read_feed("gtfs_stm.zip", dist_units="m")
     logging.info("Feed loaded")
 
-    with tempfile.TemporaryDirectory(dir=".", prefix="stop_timetable-") as d:
-        df = feed.stops
-        for _, s in df[df["stop_name"].isin(STOPS)].iterrows():
-            tt = feed.build_stop_timetable(s["stop_id"], [date.strftime("%Y%m%d")])
-            tt["stop_name"] = [s["stop_name"]] * len(tt)
-            with open(f'{d}/stop-{s["stop_id"]}.txt', "w") as f:
-                f.write(str(tt))
-            tt.to_csv(f'{d}/stop-{s["stop_id"]}.csv')
-            tt.to_json(f'{d}/stop-{s["stop_id"]}.json')
-            tt.to_pickle(f'{d}/stop-{s["stop_id"]}.pickle')
-            tt.to_html(f'{d}/stop-{s["stop_id"]}.html')
-            logging.info("Built stop %s (%s)", s["stop_id"], s["stop_name"])
+    with tempfile.TemporaryDirectory(dir=".", prefix="stop_timetable-") as temp_dir:
+        stops_df = feed.stops
+        target_stops_df = stops_df[stops_df["stop_name"].isin(STOPS)]
+
+        for _, stop_row in target_stops_df.iterrows():
+            stop_id = stop_row["stop_id"]
+            stop_name = stop_row["stop_name"]
+
+            # Build the timetable for the requested date (format: YYYYMMDD)
+            timetable = feed.build_stop_timetable(stop_id, [req_date.strftime("%Y%m%d")])
+            timetable["stop_name"] = [stop_name] * len(timetable)
```

Comment:
Hold up, earlier you wrote this with scalar on rhs:
```
tt["realtime"] = False
```

should this also be `... = stop_name` ?